src/databricks/labs/blueprint/paths.py

```diff
@@ -520,7 +520,18 @@ def resolve(self: P, strict: bool = False) -> P:
         if strict and not absolute.exists():
             msg = f"Path does not exist: {self}"
             raise FileNotFoundError(msg)
-        return absolute
+        return absolute.normalize()
+
+    def normalize(self: P) -> P:
```

Comment:
A quick initial impression while I have a closer look at this: we should not be introducing new public methods like this, they're not part of the `Path()` API being emulated here.

Reply:
done